syntax = "proto3";

message PersonMessage {
    int64 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}

message GetPersonRequest {
    int64 id = 1;
}

message GetPersonResponse {
    PersonMessage data = 1;
}

message CreatePersonRequest {
    PersonMessage payload = 1;
}

message CreatePersonResponse {
    PersonMessage data = 1;
}

message GetAllPersonRequest {}

message GetAllPersonResponse {
    repeated PersonMessage list = 1;
}

service Person {
    rpc Get(GetPersonRequest) returns (GetPersonResponse);
    rpc Create(CreatePersonRequest) returns (CreatePersonResponse);
    rpc GetAll(GetAllPersonRequest) returns (GetAllPersonResponse);
}
